packages/client/hmi-client/src/services/model-templating.ts

```diff
@@ -526,26 +542,29 @@ export function flattenedToDecomposedInView(
 	let yPos = 100;
 	const allInitals: Initial[] = [];
 
-	templatesToAdd.forEach((model: Model) => {
-		if (model.semantics?.ode?.initials) {
-			model.metadata = {
-				templateCard: {
-					id: model.header.name,
-					name: model.header.name,
-					x: 100,
-					y: yPos
-				} as ModelTemplateCard
-			};
-			yPos += 200;
+	// Make sure cards are rendered before junctions and edges (this is required to determine port positions)
+	await Promise.all(
+		templatesToAdd.map((model: Model) => {
+			if (model.semantics?.ode?.initials) {
+				model.metadata = {
+					templateCard: {
+						id: model.header.name,
+						name: model.header.name,
+						x: 100,
+						y: yPos
+					} as ModelTemplateCard
+				};
+				yPos += 200;
 
-			addTemplateInView(decomposedCanvas, model);
-			allInitals.push(...model.semantics.ode.initials);
-		}
-	});
+				addTemplateInView(decomposedCanvas, model);
+				allInitals.push(...model.semantics.ode.initials);
+			}
+			return Promise.resolve();
+		})
+	);
```

Comment:
How does this promise.all work? I don't see anything that is async in nature??

Reply:
My mistake, the addTemplate function returns a promise now. I was playing around with it,`return Promise.resolve` happened to work somehow.